### ISSUE TITLE
[RFC]: Upgrading to React 18 for Redwood

### ISSUE THREAD

**Issue by Gresliebear** [redwoodjs/redwood#5881]
### Summary

I want to use the latest package for React 18, because packages on npm install used React 18 but currently Redwood doesn't 

![image](https://user-images.githubusercontent.com/13157730/177473906-7916f893-1e40-4270-b450-7c5aba9d0e84.png)

### Motivation

1) I want to use the latest version of three.js

2) I want to use the latest version of react-three-fiber 

### Detailed proposal

Proposal I install React 18 and try running it on Redwood and make detail errors in this thread. 

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by Gresliebear** [comment]
For those who want to install react-three/fiber use this `npm i @react-three/fiber@7.0.6`

and for Three.js on redwood `npm i three@0.131.3`

**Comment by jtoar** [comment]
Thanks @Gresliebear! @virtuoushub has tried to upgrade to React 18 on and off here: https://github.com/redwoodjs/redwood/pull/4992.

Originally, we had to wait for many other packages we use to use upgrade, but I'm not so sure we're waiting on too many other packages anymore. Maybe just storybook v7. But even then, v6.5.9 may still work with react 18, it's just not written in it.

If Redwood core could support it (i.e., the router, the web side—running `yarn rw dev` and `build` work) I think it make sense to allow users to **opt into** React 18. That'd require widening our peer dependency range. 

I haven't tried in depth yet, but here's some an incomplete list of some of the things we may have to do to upgrade to React 18...

- React 18 types—are all our types still correct? I'm sure some would have to be rewritten
- auth providers—do they all support React 18? I don't think it's a hard requirement, but we'd have to document those that don't
- `yarn rw prerender`—does it still work?

Note that once we upgrade, we'd probably need to give the router a hard look. But again, I'm not sure that it's a hard requirement, just something we should do.

If you wanted to take a next step here, I'd try @virtuoushub's PR locally and see what happens.

**Comment by Gresliebear** [comment]
> 

should follow through on all these steps @jtoar mention or install React 18 and follow the latest from @virtuoushub 

I was gonna install React 18 run through the tutorial completely from the docs and document as I went.

**Comment by Gresliebear** [comment]
> auth providers—do they all support React 18? I don't think it's a hard requirement, but we'd have to document those that don't

Oh see yes auth providers and then types and then yarn rw prerender gotcha! i will get to it after work

**Comment by dthyresson** [comment]
@thedavidprice I recall that you had some info on upgrading to React 18 after coming back from Prisma Day?

**Comment by jtoar** [comment]
> should follow through on all these steps @jtoar mention or install React 18 and follow the latest from @virtuoushub
>
> I was gonna install React 18 run through the tutorial completely from the docs and document as I went.

From what @virtuoushub said, it sounds like the tutorial would work, but confirming that does indeed would certainly be valuable! And trying out a few auth providers and prerender especially too would get us closer to 100% coverage. But no pressure of course! Either way, I'll queue up React 18 for discussion cause it sounds like we could let people opt into it a lot sooner than I thought.

**Comment by Gresliebear** [comment]
tutorial React 18 errrors 

cd into E:\RedwoodWindows\redwood\packages\web framework 

uninstalled react and react-dom in the web packages.json
`yarn remove react `
`yarn remove react-dom `

my test project when I 

`yarn cross-env RWFW_PATH=E:\RedWoodWindows\redwood rwfw project:sync` seems to be work correctly however 

now in my test project when I run `yarn run dev`

![image](https://user-images.githubusercontent.com/13157730/178868264-3e85ecc9-8771-4717-a372-ad5d87169c5b.png)

**Comment by Gresliebear** [comment]
@virtuoushub it will take me a little bit to catch up to these changes on my end I just got createRoot and rootHydrate working https://github.com/redwoodjs/redwood/pull/4992/commits

**Comment by Gresliebear** [comment]
### Continuing Tutorial 

`yarn redwood generate page about`

works

Chapter 1 works in general with React 18 updates and CreateRoot

**Comment by adriatic** [comment]
A quick comment to observation made by @Gresliebear [above](https://github.com/redwoodjs/redwood/issues/5881#issuecomment-1183857875)

![image](https://user-images.githubusercontent.com/2712405/191845533-23e7d4c0-4965-4424-b7f0-9b5e83738ff4.png)

I made the same mistake too many times: the command `yarn run dev` is incorrect - it should be `yarn rw dev` 😄, so all is cool there.

P.S. I visited this issue because of the @virtuoushub's remark "kudos to @Gresliebear for all work done on testing" - so I am joining Peter.
